mmdet/apis/ote/apis/detection/nncf_task.py

```diff
@@ -0,0 +1,520 @@
+
+import numpy as np
+
+from ote_sdk.entities.inference_parameters import InferenceParameters
+from ote_sdk.entities.metrics import (CurveMetric,
+                                      LineChartInfo,
+                                      MetricsGroup,
+                                      Performance,
+                                      ScoreMetric,
+                                      InfoMetric,
+                                      VisualizationType,
+                                      VisualizationInfo)
+from ote_sdk.entities.shapes.box import Box
+from ote_sdk.entities.optimization_parameters import OptimizationParameters
+from ote_sdk.entities.label import ScoredLabel
+
+from sc_sdk.configuration import cfg_helper, ModelConfig
+from ote_sdk.configuration.helper.utils import ids_to_strings
+from sc_sdk.entities.annotation import Annotation
+from sc_sdk.entities.datasets import Dataset, Subset
+from sc_sdk.entities.optimized_model import OptimizedModel, ModelPrecision
+from sc_sdk.entities.task_environment import TaskEnvironment
+
+
+from sc_sdk.entities.model import Model, ModelStatus, NullModel
+
+from sc_sdk.entities.resultset import ResultSet, ResultsetPurpose
+from sc_sdk.usecases.evaluation.metrics_helper import MetricsHelper
+from sc_sdk.usecases.reporting.time_monitor_callback import TimeMonitorCallback
+from sc_sdk.usecases.tasks.interfaces.evaluate_interface import IEvaluationTask
+from sc_sdk.usecases.tasks.interfaces.inference_interface import IInferenceTask
+from sc_sdk.usecases.tasks.interfaces.optimization_interface import IOptimizationTask, OptimizationType
+from sc_sdk.usecases.tasks.interfaces.export_interface import IExportTask, ExportType
+from sc_sdk.usecases.tasks.interfaces.unload_interface import IUnload
+from sc_sdk.logging import logger_factory
+
+from mmcv.parallel import MMDataParallel
+from mmcv.runner import load_checkpoint
+from mmcv.utils import Config
+from mmdet.apis import train_detector, single_gpu_test, export_model
+from mmdet.apis.ote.apis.detection.config_utils import patch_config
+from mmdet.apis.ote.apis.detection.config_utils import set_hyperparams
+from mmdet.apis.ote.apis.detection.config_utils import prepare_for_training
+from mmdet.apis.ote.apis.detection.config_utils import prepare_for_testing
+from mmdet.apis.ote.apis.detection.configuration import OTEDetectionConfig
+from mmdet.apis.ote.extension.utils.hooks import OTELoggerHook
+from mmdet.apis.train import create_nncf_model
+from mmdet.datasets import build_dataset, build_dataloader
+from mmdet.integration.nncf import check_nncf_is_enabled
+from mmdet.integration.nncf import wrap_nncf_model
+from mmdet.integration.nncf.config import compose_nncf_config
+from mmdet.models import build_detector
+from mmdet.parallel import MMDataCPU
+
+logger = logger_factory.get_logger("OTEDetectionTask")
+
+
+class NNCFDetectionTask(IOptimizationTask, IInferenceTask, IExportTask, IEvaluationTask, IUnload):
```

Comment:
Should not we inherit from the Task implemented in task.py as the first step to avoid code duplication?